test/e2e/replica_set_custom_persistent_volume/replica_set_custom_persistent_volume.go

```diff
@@ -0,0 +1,141 @@
+package replica_set_custom_persistent_volume
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	v1 "github.com/mongodb/mongodb-kubernetes-operator/api/v1"
+	"github.com/mongodb/mongodb-kubernetes-operator/test/e2e/mongodbtests"
+	. "github.com/mongodb/mongodb-kubernetes-operator/test/e2e/util/mongotester"
+	"github.com/stretchr/testify/assert"
+
+	e2eutil "github.com/mongodb/mongodb-kubernetes-operator/test/e2e"
+	setup "github.com/mongodb/mongodb-kubernetes-operator/test/e2e/setup"
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestMain(m *testing.M) {
+	code, err := e2eutil.RunTest(m)
+	if err != nil {
+		fmt.Println(err)
+	}
+	os.Exit(code)
+}
+
+func getPersistentVolumeLocal(name string, localPath string, label string) corev1.PersistentVolume {
+	return corev1.PersistentVolume{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: e2eutil.OperatorNamespace,
+			Labels:    map[string]string{"type": label},
```

Comment:
Use `labels` as a way to then use `matchlabels` in the PVC to bind to this specific volume